### DISCUSSION TITLE
Issue : Legacy Extension 'org.wildfly.extension.microprofile.metrics.smallrye' is not supported on servers running this version. The extension is only supported for use by hosts running a previous release in a mixed-version managed domain.

### DISCUSSION THREAD

**Discussion by ChasingDeveloper** [eclipse/microprofile-fault-tolerance#624] [Q&A]
### The existing application is on Wildfly 26.1.0.final and Jarkata EE 8 and we have existing extension and subsystem configured in cli.orig file as below: 

```
/extension=org.widlfly.extension.microprofile.fault-tolerance-smallrye:add
/extension=org.wildfly.extension.microprofile.metrics-smallrye:add
/subsystem=microprofile-fault-tolerance-smallrye:add
/subsystem=microprofile-metrics-smallrye:add
```

top level Pom - 

```
<dependency>
<groupId>org.wildfly.bom</groupId>
<artifactId>wildfly-jakartaee8</artifactId>
</dependency>

<dependency>
<groupId>org.wildfly.bom</groupId>
<artifactId>wildfly-microprofile</artifactId>
</dependency>

<dependency>
<groupId>org.eclipse.microprofile.fault-tolerance</groupId>
<artifactId>microprofile-fault-tolerance-api</artifactId>
</dependency>
```

_________________________________________________________________________________________________________

### we upgraded to Wildfly 31.0.1.Final & Jakarta EE 10 and changed the existing pom as below:

top level Pom - 

```
<dependency>
<groupId>org.wildfly.bom</groupId>
<artifactId>wildfly-ee-with-tools</artifactId>
</dependency>

<dependency>
<groupId>org.wildfly.bom</groupId>
<artifactId>wildfly-microprofile</artifactId>
</dependency>

<dependency>
<groupId>org.eclipse.microprofile.fault-tolerance</groupId>
<artifactId>microprofile-fault-tolerance-api</artifactId>
</dependency>
```

When deploying into server with new changes - 
Server boot has failed in an unrecoverable manner and not starting.

**Caused by: Legacy Extension 'org.wildfly.extension.microprofile.metrics.smallrye' is not supported on servers running this version. The extension is only supported for use by hosts running a previous release in a mixed-version managed domain.**

After research i found some information below from https://docs.wildfly.org/wildfly-proposals/microprofile/WFLY-11529_subsystem_metrics.html
<img width="1090" alt="Screenshot 2024-04-01 at 4 29 09 PM" src="https://github.com/eclipse/microprofile-fault-tolerance/assets/90581421/b49afe7d-a5d2-4488-93e9-a0d622df46ca">

I'm not sure can i find workaround to fix this issue. 

Could anyone kindly help me if anyone faced this issue and where should i find any alternatives to fix this problem. Any leads are greatly appreciated. Thanking you in anticipation.

**Comment by Ladicek** [comment]
Remove the SmallRye Metrics extension, WildFly no longer supports that. Instead, add a Micrometer extension.

Also, this is not actually related to MicroProfile Fault Tolerance, so closing.
